include/deal.II/base/index_set.h

```diff
@@ -1654,9 +1691,34 @@ IndexSet::add_indices(const ForwardIterator &begin, const ForwardIterator &end)
           ++q;
         }
 
-      add_range(begin_index, end_index);
+      tmp_ranges.emplace_back(begin_index, end_index);
       p = q;
+      if (p != end && *p < end_index)
+        ranges_are_sorted = false;
```

Comment:
Can you add a comment in front of the `if` here? I think what you're saying is that the starting index of the range we will identify in the next go-around of the `for` loop is less than the end index of the one just identified, then we will have at least one pair of ranges that are not sorted, and consequently the whole collection of ranges is not sorted.